### ISSUE TITLE
Continue "asset amount and value" graph after the amount goes to zero

### ISSUE THREAD

**Issue by rkalis** [rotki/rotki#2822]
There are a few assets in my Rotki that I used to have, but I now don't have. When looking "asset amount and value" graph it ends at the last non-zero amount, and it states under the graph that the data is available until the date I sold the asset. It would make sense to me if the graph continued on until the current date, but the amount / value goes to zero.

**Comment by LefterisJP** [comment]
Hey @rkalis thanks for opening an issue to discuss this!

The logic there is quite simple. We don't save zero balances in the DB snapshots (which imo is good). And when getting the data for plotting out of the DB we get the snapshots for which the user has data and plot them.

So what you are asking for is that if there is snapshot data for an asset at any point in time, make the graph continue until now even if the data ends in the past, showing a 0 value for between data end and now?

Is my understanding correct?

**Comment by rkalis** [comment]
Yes exactly. That way it's clearer at a glance that you're not currently holding the asset, rather than looking at the date for that.

**Comment by LefterisJP** [comment]
@kelsos @lukicenturi do you think this issue makes sense?

For example for such a graph, as below to show it until today, just with auto-inserted zero values?

![2022-05-21_10-17](https://user-images.githubusercontent.com/1658405/169642768-fa496666-5460-4a85-a875-09d61ea2fbcc.png)

If yes do you think it makes sense to do it in the frontend or the backend?

I am gonna move it to another random milestone but we should try to just implement to get it off the tracker.

**Comment by kelsos** [comment]
For the front to do it we should at least when went it went to zero right? at the moment we don't really know

**Comment by LefterisJP** [comment]
> For the frontend to do it we should at least when went it went to zero right? at the moment we don't really know since there is no value stored for 0 or missing.

So @kelsos for this wouldn't knowing the last positive value be enough? So if last value is positive and after that there is a zero you can assume all in-between is zeros right?

**Comment by kelsos** [comment]
Yes, pretty much that was what I had in mind, just save one zero value whenever an asset goes to zero and assume that everything after that is zero until the next positive value.

This way you don't have unnecessary zero values in the snapshots. You save zero only at the start of the period where it is zero and unless it becomes positive you never save a snapshot for the asset again.

This way the frontend can assume that anything after zero until a positive value appears is zero and inject the values in the frontend graph.

**Comment by LefterisJP** [comment]
So with my current understanding of how this can be implemented we need to do the following in the backend, which I find a bit ugly:

1. To make sure all snapshots from here and on are okay, whenever a new balances snapshot is taken we need to be checking the previous one and if any asset had positive value in the last one and no value in this one, add a zero. This will increase our snapshots size (a tiny bit)
2. To make this work historically though, we need a data migration that will run through all snapshots and perform the same logic historically. This will increase the user DB ( a bit).

Writing it down and reading it, it sounds really hacky and adding artificial values where there should be none, inflating the DB size for nothing. I don't like it and think it's not a good approach.

We could perhaps achieve the same by a smarter sqlite query for the data shown in the per asset value graph? So get the asset (values,times) but also check if there is any snapshots where there is no value for this asset and fill in the blanks only at the api level when returning.

Super simple example:

GNO snapshots (value, time):
`(1, 1), (1, 2), (2, 3), (5, 6)`

All snapshots timestamps:
`(1, 2, 3, 4, 5, 6,)`

Requesting GNO snapshot value data via api:
So the api would see there is a snapshot for timestamp `4` and `5` and that this is in between `3` and `6` which have values in the snapshot. So it would return that extra 0 Kelsos want like this:
`(1, 1), (1, 2), (2, 3),(0, 4),(5, 6)`

**Comment by kelsos** [comment]
As we discussed during the standup if you can infer the value, that would be fine too.
